_literate/EX-boston-flux.jl

```diff
@@ -0,0 +1,128 @@
+# **Main author**: Ayush Shridhar (ayush-1506).
+#
+# ## Getting started
+
+using MLJFlux, MLJ, DataFrames, Statistics, Flux
+
+MLJ.color_off() # hide
+Random.seed!(11)
+
+# Loading the Boston dataset. Our aim will be to implement a
+# neural network regressor to predict the price of a house, 
+# given a number of features.
+
+features, targets = @load_boston
+features = DataFrame(features)
+@show size(features)
+@show targets[1:3]
+first(features, 3) |> pretty
+
+# Next obvious steps: partitioning into train and test set
+
+train, test = partition(eachindex(targets), 0.70, shuffle=true, rng=52)
+
+# Let us try to implement an Neural Network regressor using
+# Flux.jl. MLJFlux.jl provides an MLJ interface to the Flux.jl
+# deep learning framework. The package provides four essential
+# models: `NeuralNetworkRegressor, MultivariateNeuralNetworkRegressor,
+# NeuralNetworkClassifier` and `ImageClassifier`. 
+
+# At the heart of these models is a neural network. This is specified using
+# the `builder` parameter. Creating a builder object consists of two steps:
+# 1. Creating a new struct inherited from `MLJFlux.Builder`. `MLJFlux.Builder`
+# is an abstract structure used for the purpose of dispatching. Suppose we define
+# a new struct called `MyNNRegressor`. This can contain any attribute required to
+# build the model later. (Step 2). Let's use Dense Neural Network with 2 hidden layers.
+
+mutable struct MyNNRegressor <: MLJFlux.Builder
+    n1::Int #Number of cells in the first hidden layer
+    n2::Int #Number of cells in the second hidden layer
+end
```

Comment:
I think it's confusing to name the builder object "MyNNRegressor". In principle the same builder could be used for classification, and so forth. How about "MyBuilder" or "MyNetworkBuilder"?